pkg/requests/builder.go

```diff
@@ -40,8 +42,23 @@ func New(endpoint string) Builder {
 
 // WithContext adds a context to the request.
 // If no context is provided, context.Background() is used instead.
+// Will propagate the tracing headers from the context automatically.
 func (r *builder) WithContext(ctx context.Context) Builder {
 	r.context = ctx
+	return r.injectTracingHeaders(ctx)
```

Comment:
Is this needed?

The request builder is used for sending subrequests to different Identity Providers during the various authentication subrequests. I would almost think we don't want to leak internal tracing data to external 3rd party identity providers?

The upstream proxy gets sent the original request, so if you add tracing support to your context in the middleware, I think it should in the context of the request we will be proxying to the upstream.

Reply:
That's a good point and why I think it should be disabled by default.

This tracing is useful when e.g. istio intercepts the egress traffic and strips the tracing headers, but uses them to trace the outgoing request itself, thus giving you insight into the spans outside of your network.

Reply:
Right - my point was this code path isn't used for proxied requests, your PR mentioned you are looking for tracing data on requests to your upstream application?

The request builder is used for hitting the external provider Auth, Redeem & Userinfo endpoints almost exclusively.

Reply:
Oh sorry, I follow your point -- you get the ability to see this with Istio even though it is to an external entity because of its network mesh. Am I following that correctly now?

Reply:
> you get the ability to see this with Istio even though it is to an external entity because of its network mesh

exactly. Besides, in the istio case oauth2-proxy isn't used as a proxy itself, only as an auth mechanism.